aptos-move/aptos-vm/src/sharded_block_executor/sharded_aggregator_service.rs

```diff
@@ -0,0 +1,308 @@
+                        DeltaU128::get_delta(
+                            last_txn_total_supply.unwrap(),
+                            TOTAL_SUPPLY_AGGR_BASE_VAL,
+                        );
+                    break;
+                }
+            }
+        });
+
+    for idx in 1..aggr_total_supply_delta.len() - 1 {
+        aggr_total_supply_delta[idx + 1] =
+            aggr_total_supply_delta[idx + 1] + aggr_total_supply_delta[idx];
+    }
+
+    // The txn_outputs contain 'txn_total_supply' with
+    // 'CrossShardStateViewAggrOverride::total_supply_aggr_base_val' as the base value.
+    // The actual 'total_supply_base_val' is in the state_view.
+    // The 'delta' for the shard/round is in aggr_total_supply_delta[round * num_shards + shard_id + 1]
+    // For every txn_output, we have to compute
+    //      txn_total_supply = txn_total_supply - CrossShardStateViewAggrOverride::total_supply_aggr_base_val + total_supply_base_val + delta
+    // While 'txn_total_supply' is u128, the intermediate computation can be negative. So we use
+    // DeltaU128 to handle any intermediate underflow of u128.
+    let total_supply_base_val: u128 = get_state_value(&TOTAL_SUPPLY_STATE_KEY, state_view).unwrap();
+    let base_val_delta = DeltaU128::get_delta(total_supply_base_val, TOTAL_SUPPLY_AGGR_BASE_VAL);
+
+    let aggr_total_supply_delta_ref = &aggr_total_supply_delta;
+    // Runtime is O(num_txns), hence parallelized at the shard level and at the txns level.
+    executor_thread_pool.scope(|s| {
+        for (shard_id, shard_output) in sharded_output.iter_mut().enumerate() {
+            s.spawn(move |_| {
+                for (round, txn_outputs) in shard_output.iter_mut().enumerate() {
+                    let delta_for_round =
+                        aggr_total_supply_delta_ref[round * num_shards + shard_id] + base_val_delta;
+                    let total_txns_len = txn_outputs.len();
+                    for (txn_idx, txn_output) in txn_outputs.iter_mut().enumerate() {
+                        let txn_total_supply = txn_output.write_set().get_total_supply();
+                        if is_total_supply_absent(
+                            txn_total_supply,
+                            shard_id,
+                            num_shards,
+                            txn_idx,
+                            total_txns_len,
+                        ) {
+                            continue;
+                        }
+                        txn_output.update_total_supply(
+                            delta_for_round.add_delta(txn_total_supply.unwrap()),
+                        );
+                    }
+                }
+            });
+        }
+    });
+
+    let delta_for_global_shard = aggr_total_supply_delta[num_shards * num_rounds] + base_val_delta;
+    let delta_for_global_shard_ref = &delta_for_global_shard;
+    let global_output_len = global_output.len();
+    if global_output.len() < 25 {
```

Comment:
You don't need this.. just use `iter_mut`  with min length..

Reply:
Ok done